src/core/packages/http/router-server-internal/src/router.ts

```diff
@@ -183,8 +184,20 @@ export class Router<Context extends RequestHandlerContextBase = RequestHandlerCo
   public registerRoute(route: InternalRouterRoute) {
     this.routes.push({
       ...route,
-      handler: async (request, responseToolkit) =>
-        await this.handle({ request, responseToolkit, handler: route.handler }),
+      handler: async (request, responseToolkit) => {
+        /**
+         * Read incoming traceparent headers and create a new context with the traceparent set.
+         * This allows OpenTelemetry spans created in the next context to re-use the traceparent
+         * headers (and thus belonging to the same trace). It does not interfere with Elastic APM,
+         * and is temporary until we fully migrate to [OpenTelemetry
+         * tracing](https://github.com/elastic/kibana/issues/220914).
+         */
+        const ctx = propagation.extract(context.active(), request.headers, defaultTextMapGetter);
```

Comment:
I wonder if it's worth moving this to a previous handler. This way, the context can be reused for any previous lifecycle steps like authentication/authorization, validation, etc.

I don't think that it should block the PR, though. We can move it in a follow-up PR.

WDYT?

Reply:
yes makes sense: https://github.com/elastic/kibana/issues/225817

Reply:
sorry, replied to the wrong comment - wdym with:
> I wonder if it's worth moving this to a previous handler.

It should definitely happen as early as possible. where would I need to put it then?

Reply:
ah that does not actually work I think, you need to be able to wrap a callback